packages/rancher-sachet/charts/templates/configmap-pre-install.yaml

```diff
@@ -0,0 +1,21 @@
+{{/*This file is applied when the operation is helm install and the target confimap does not exist. */}}
+{{- if not (lookup "v1" "ConfigMap" (include "sachet.namespace" . ) (include "sachet.fullname" .)) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: {{ include "sachet.namespace" . }}
+  name: {{ include "sachet.fullname" . }}
+  labels: {{ include "sachet.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "3"
+    "helm.sh/resource-policy": keep
+data:
+  config.yaml: |-
+    {{- with .Values.sachet.providers }}
+    providers: {{ toYaml . | nindent 6 }}
+    {{- end }}
+    {{- with .Values.sachet.receivers }}
+    receivers: {{ toYaml . | nindent 6 }}
+    {{- end }}
```

Comment:
what happens if both are empty?

Reply:
in that case, the app will not be installed at all because of the following check in `packages/rancher-alerting-drivers/charts/templates/cluster-role.yaml`
```
{{- if and (not .Values.sachet.enabled) (not .Values.prom2teams.enabled) -}}
{{- fail "At least one Driver must be enabled to install the chart. " }}
{{- end -}}
```

Reply:
What if `.Values.sachet.enabled=true` but `.Values.sachet.providers` and `.Values.sachet.receivers` are empty though? Maybe I'm missing something here.

Reply:
Both `.Values.sachet.providers` and `.Values.sachet.receivers` are emtpy by default, the configMap conains the config.yaml with no content. Like this:
```
---
# Source: rancher-alerting-drivers/charts/sachet/templates/configmap-pre-install.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: default
  name: test1-sachet
  labels: 
    helm.sh/chart: sachet-1.0.1
    app.kubernetes.io/name: sachet
    app.kubernetes.io/instance: test1
    app.kubernetes.io/version: "0.2.3"
    app.kubernetes.io/managed-by: Helm
  annotations:
    "helm.sh/hook": pre-install
    "helm.sh/hook-weight": "3"
    "helm.sh/resource-policy": keep
data:
  config.yaml: |-
```
Users need to go to edit the configMap to put real values.

Reply:
Is leaving a |- without any additional line valid YAML? e.g. does a helm install with this ConfigMap work?

If so, we can resolve this comment and move forward

Reply:
yes, it works.